k9mail/src/main/java/com/fsck/k9/notification/NotificationController.java

```diff
@@ -102,7 +103,13 @@ public void clearFetchingMailNotification(Account account) {
     }
 
     public void addNewMailNotification(Account account, LocalMessage message, int previousUnreadMessageCount) {
-        newMailNotifications.addNewMailNotification(account, message, previousUnreadMessageCount);
+        boolean hasHighPriority = false;
+
+        for (String iter : message.getHeader("X-Priority")) {
```

Comment:
this logic should be part of `Message`. it's also unnecessary to check all headers, just the first one should suffice.